state/machine_test.go

```diff
@@ -170,96 +172,120 @@ func (s *MachineSuite) testSetUpgradeCharmProfileWithoutLXDProfile(c *gc.C, prof
-
-	assertUpgradeCharmProfileRequired(c, m, "lxd-profile", "local:quantal/quantal-lxd-profile-0")
-
-	err = m.SetUpgradeCharmProfile("lxd-profile", "local:quantal/quantal-lxd-profile-0")
+	err := s.machine.SetUpgradeCharmProfile(s.unit.Name(), "local:quantal/quantal-lxd-profile-0")
 	c.Assert(err, jc.ErrorIsNil)
 }
 
 func (s *MachineSuite) TestSetUpgradeCharmProfileWithLXDProfile(c *gc.C) {
-	m := s.machine
-
-	app := s.AddTestingApplication(c, "lxd-profile", s.AddTestingCharm(c, "lxd-profile"))
-	unit, err := app.AddUnit(state.AddUnitParams{})
-	c.Assert(err, jc.ErrorIsNil)
-	err = unit.AssignToMachine(m)
-	c.Assert(err, jc.ErrorIsNil)
-
-	err = m.SetUpgradeCharmProfile("lxd-profile", "local:quantal/quantal-lxd-profile-0")
-	c.Assert(err, jc.ErrorIsNil)
-
-	assertUpgradeCharmProfileRequired(c, m, "lxd-profile", "local:quantal/quantal-lxd-profile-0")
+	s.assertSetUpUnitSetUpgradeCharmProfile(c)
+	s.assertUpgradeCharmProfileRequired(c, s.unit.Name(), "local:quantal/quantal-lxd-profile-0")
 }
 
 func (s *MachineSuite) TestSetUpgradeCharmProfileComplete(c *gc.C) {
+	s.assertSetUpUnitSetUpgradeCharmProfile(c)
 	m := s.machine
 
-	app := s.AddTestingApplication(c, "lxd-profile", s.AddTestingCharm(c, "lxd-profile"))
-	unit, err := app.AddUnit(state.AddUnitParams{})
-	c.Assert(err, jc.ErrorIsNil)
-	err = unit.AssignToMachine(m)
-	c.Assert(err, jc.ErrorIsNil)
-
-	err = m.SetUpgradeCharmProfile("app-name", "local:quantal/quantal-lxd-profile-0")
-	c.Assert(err, jc.ErrorIsNil)
-
-	err = m.SetUpgradeCharmProfileComplete(lxdprofile.SuccessStatus)
+	err := m.SetUpgradeCharmProfileComplete(s.unit.Name(), lxdprofile.SuccessStatus)
 	c.Assert(err, jc.ErrorIsNil)
 
 	err = m.Refresh()
 	c.Assert(err, jc.ErrorIsNil)
 
-	status, err := m.UpgradeCharmProfileComplete()
+	status, err := m.UpgradeCharmProfileComplete(s.unit.Name())
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(status, gc.Equals, lxdprofile.SuccessStatus)
 }
 
-func assertUpgradeCharmProfileNotRequired(c *gc.C, m *state.Machine, expectedAppName string) {
-	err := m.Refresh()
-	c.Assert(err, jc.ErrorIsNil)
+func (s *MachineSuite) TestNextUpgradeCharmProfileApplicationName(c *gc.C) {
+	s.assertSetUpUnitSetUpgradeCharmProfile(c)
 
-	obtainedAppName, err := m.UpgradeCharmProfileApplication()
+	name, err := s.machine.NextUpgradeCharmProfileUnitName()
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(obtainedAppName, gc.Equals, expectedAppName)
-	charmURL, err := m.UpgradeCharmProfileCharmURL()
+	c.Assert(name, gc.Equals, s.unit.Name())
+}
+
+func (s *MachineSuite) TestNextUpgradeCharmProfileApplicationNameZero(c *gc.C) {
+	s.assertSetUpUnitForLXDProfile(c)
+	m := s.machine
+
+	name, err := m.NextUpgradeCharmProfileUnitName()
+	c.Assert(err, gc.ErrorMatches, "cannot get instance charm profile data for machine 1: instance charm profile data for machine 1 not found")
+	c.Assert(name, gc.Equals, "")
+}
+
+//func (s *MachineSuite) TestSingleUpgradeCharmProfileApplicationNameMoreThanOne(c *gc.C) {
```

Comment:
Any reason why this is commented out?

Reply:
this was related to only returning a name from  NextUpgradeCharmProfileUnitName if 1 doc existed.  that functionality has been changed to allow more than one.  Added a test case to cover the change